### ISSUE TITLE
[Static Shock] - I want to load menu configurations

### ISSUE THREAD

**Issue by matthew-carroll** [Flutter-Bounty-Hunters/static_shock#14]
I'm building the "Guides" page for the Docs website. The Guides page has a side-menu of buttons, which should display different pieces of content in the primary content area.

Currently, I'm hard-coding these menu buttons into the .jinja layout for the page. This technically solves the problem, but it requires layout editing to add, remove, or update one of those menu items.

Consider making it possible to source menu items from some other place, and make it available as page data.

This problem is not specific to menu items. It could be true for any data that a user might want to update on a page without editing the HTML layout. This is especially true when the page is defined through Markdown, instead of through a direct .jinja template.

**Comment by matthew-carroll** [comment]
I've now moved the menu configuration into each page's Front Matter. This still suffers the same problem. I moved the config to YAML to fit into the template that's used for all guide pages.

What we probably want is a system similar to Lume's, which lets users define arbitrary data within any given directory scope, and that data is inherited down to deeper scopes. Regardless, we want the ability to inject arbitrary data in some manner.
